json-rpc/src/transactions.rs

```diff
@@ -0,0 +1,148 @@
+                    timestamp_usecs: t.get_timestamp_usecs(),
+                }),
+                writeset: None,
+                user: None,
+            }),
+            Transaction::WriteSet(_) => Ok(TransactionData {
+                blockmetadata: None,
+                writeset: Some(WriteSetTransactionView {}),
+                user: None,
+            }),
+
+            Transaction::UserTransaction(t) => {
+                let empty_vec: Vec<TransactionArgument> = vec![];
+                let payload = t.clone().into_raw_transaction().into_payload();
+                let (code, args) = match payload {
+                    TransactionPayload::Program => ("deprecated".to_string(), empty_vec),
+                    TransactionPayload::WriteSet(_) => ("genesis".to_string(), empty_vec),
+                    TransactionPayload::Script(script) => {
+                        (get_transaction_name(script.code()), script.args().to_vec())
+                    }
+                    TransactionPayload::Module(_) => ("module publishing".to_string(), empty_vec),
+                };
+
+                Ok(TransactionData {
+                    blockmetadata: None,
+                    writeset: None,
+                    user: Some(UserTransactionView {
+                        sender: t.sender().to_string(),
+                        signature: t.signature().to_string(),
+                        public_key: t.public_key().to_string(),
+                        sequence_number: t.sequence_number(),
+                        max_gas_amount: t.max_gas_amount(),
+                        gas_unit_price: t.gas_unit_price(),
+                        expiration_time: t.expiration_time().as_secs(),
+                        script_name: code.clone(),
+                        script_args: UserTranscationArgs {
+                            peer_to_peer: if code == "peer_to_peer" {
+                                if let [TransactionArgument::Address(receiver), TransactionArgument::U64(amount)] =
+                                    &args[..]
+                                {
+                                    Some(PeerToPeerArgs {
+                                        receiver: receiver.to_string(),
+                                        amount: *amount,
+                                    })
+                                } else {
+                                    bail!("Unable to parse peer_to_peer arguments")
+                                }
+                            } else {
+                                None
+                            },
+                        },
+                    }),
+                })
+            }
+        }
+    }
+}
+
+/// Returns transactions by range
+pub(crate) async fn get_transactions(
+    libra_db: Arc<LibraDB>,
+    params: Vec<Value>,
+) -> Result<Vec<TransactionView>> {
+    let p_start_version: Number = serde_json::from_value(params[0].clone())?;
```

Comment:
just curious - are you specifying the types of variables because the compiler was complaining? or ease of reading code? usually the compiler should be smart enough so that it can infer the variable types

Reply:
from_value require an Type, otherwise it doesn't know the type it converting to